diff --git a/go.mod b/go.mod
index bba11c1..bacc4b1 100644
--- a/go.mod
+++ b/go.mod
@@ -1,10 +1,10 @@
 module github.com/fmitra/srp
 
+go 1.22.4
+
+require github.com/stretchr/testify v1.2.2
+
 require (
-	github.com/BurntSushi/toml v0.3.1 // indirect
-	github.com/davecgh/go-spew v1.1.1
-	github.com/pmezard/go-difflib v1.0.0
-	github.com/stretchr/testify v1.2.2
-	golang.org/x/tools v0.0.0-20190208222737-3744606dbb67 // indirect
-	honnef.co/go/tools v0.0.0-20190128043916-71123fcbb8fe // indirect
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
 )
diff --git a/go.sum b/go.sum
index 5020a95..e03ee77 100644
--- a/go.sum
+++ b/go.sum
@@ -1,12 +1,6 @@
-github.com/BurntSushi/toml v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
-github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
 github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=
 github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
 github.com/stretchr/testify v1.2.2 h1:bSDNvY7ZPG5RlJ8otE/7V6gMiyenm9RtJ7IUVIAoJ1w=
 github.com/stretchr/testify v1.2.2/go.mod h1:a8OnRcib4nhh0OaRAV+Yts87kKdq0PP7pXfy6kDkUVs=
-golang.org/x/tools v0.0.0-20190208222737-3744606dbb67 h1:bPP/rGuN1LUM0eaEwo6vnP6OfIWJzJBulzGUiKLjjSY=
-golang.org/x/tools v0.0.0-20190208222737-3744606dbb67/go.mod h1:n7NCudcB/nEzxVGmLbDWY5pfWTLqBcC2KZ6jyYvM4mQ=
-honnef.co/go/tools v0.0.0-20190128043916-71123fcbb8fe h1:/GZ/onp6W295MEgrIwtlbnxmFSKGavFp7/D7tMVyuaM=
-honnef.co/go/tools v0.0.0-20190128043916-71123fcbb8fe/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
diff --git a/srp.go b/srp.go
index cc4ce27..90089f9 100644
--- a/srp.go
+++ b/srp.go
@@ -70,6 +70,12 @@ func (s *SRP) ephemeralPrivate() *big.Int {
 	return s.ephemeralPrivateKey
 }
 
+func (s *SRP) Pad(b []byte) []byte {
+	out := make([]byte, s.N.BitLen()>>3)
+	copy(out[len(out)-len(b):], b)
+	return out
+}
+
 // multiplierParm returns a multipler paramenter K
 // RFC 5054 2.5.3 Defines K as SHA1(N | G)
 func (s *SRP) multiplierParam() (*big.Int, error) {
@@ -83,7 +89,7 @@ func (s *SRP) multiplierParam() (*big.Int, error) {
 
 	h := s.H.New()
 	h.Write(s.N.Bytes())
-	h.Write(s.G.Bytes())
+	h.Write(s.Pad(s.G.Bytes()))
 
 	k := &big.Int{}
 	s.k = k.SetBytes(h.Sum(nil))
@@ -94,8 +100,8 @@ func (s *SRP) multiplierParam() (*big.Int, error) {
 // RFC 5054 2.5.3 Defines U as SHA1(A | B)
 func (s *SRP) scramblingParam(a, b *big.Int) *big.Int {
 	h := s.H.New()
-	h.Write(b.Bytes())
-	h.Write(a.Bytes())
+	h.Write(s.Pad(a.Bytes()))
+	h.Write(s.Pad(b.Bytes()))
 
 	u := &big.Int{}
 	s.u = u.SetBytes(h.Sum(nil))
